ogre2/src/Ogre2Scene.cc

```diff
@@ -1554,6 +1570,60 @@ bool Ogre2Scene::SkyEnabled() const
   return this->dataPtr->skyEnabled;
 }
 
+//////////////////////////////////////////////////
+void Ogre2Scene::SetShadowTextureSize(LightType _lightType,
+    unsigned int _textureSize)
+{
+  // If _textureSize exceeds max possible tex size, then use default
+  if (_textureSize > this->dataPtr->maxTexSize / 2)
```

Comment:
My `maxTexSize` is 32K, and I allow texsizes (that are pow of 2 vals) up to 16K. If I set `_textureSize` to 32K this happens:
![depot what happen if go above 16k](https://github.com/user-attachments/assets/a3c633bf-e292-4d4f-b55f-4395fa02d708)
I think maybe the math on `atlasResolution` goes crazy. (doesn't fit within atlas?)

Reply:
I see. I tested with 16K max tex size as that's what my machine supports and it works fine. So how about instead of dividing by 2, we set a hard cap on the max tex size to 16K? I remembered and found [this comment](https://github.com/gazebosim/gz-rendering/issues/187#issuecomment-890598436) mentioning the 16K texture size cap as well. 

You can do this above when setting this->dataPtr->maxTexSize, e.g.

```cpp
this->dataPtr->maxTexSize = std::min(glMaxTexSize, 16384u)
```

and add a todo note mentioning that there are issues with dir shadows when setting to a number larger than 16K.

Reply:
Capped `maxTexSize` to 16K in ddde4d9a9686780b93296a1f79658540be698cb0, added TODO comment in a8a3b65d5891e4f4a50d57c4d3665f54fbd0a5c2